From e7ae4ceed171a8c5dc166ca114ec0ae520303c72 Mon Sep 17 00:00:00 2001
From: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
Date: Tue, 17 Jul 2018 16:35:40 +0200
Subject: [PATCH]  PMCPESYS-51: thermal: ltq: Don't hardcode no. of sensor
 cells

Use standard #thermal-sensor-cells property to read out number of sensor
cells. Remove hardcodded number of cells per SoC.

Note,  #thermal-sensor-cells=<0> means 1 sensor, <1> is for 2 sensors
(0 and 1 id), etc.

Signed-off-by: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
---
 drivers/thermal/ltq_thermal.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/thermal/ltq_thermal.c b/drivers/thermal/ltq_thermal.c
index c5cde1733e5c..63a14b7922f0 100644
--- a/drivers/thermal/ltq_thermal.c
+++ b/drivers/thermal/ltq_thermal.c
@@ -35,7 +35,6 @@ struct ltq_thermal_sensor;
 struct ltq_thermal_tsens_data {
 	void (*init)(struct platform_device *pdev, struct ltq_thermal *p);
 	int (*get_temp)(struct ltq_thermal_sensor *s);
-	u8 sensors_count;
 };
 
 struct ltq_thermal_sensor {
@@ -159,7 +158,6 @@ int ltq_grx500_get_temp(struct ltq_thermal_sensor *sensor)
 static struct ltq_thermal_tsens_data ltq_grx500_data = {
 	.init		= ltq_grx500_init,
 	.get_temp	= ltq_grx500_get_temp,
-	.sensors_count	= 2,
 };
 
 static int ltq_thermal_get_temp(void *data, int *temp)
@@ -262,7 +260,12 @@ static int ltq_thermal_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->chiptop);
 	}
 
-	priv->count = pdata->sensors_count;
+	if (device_property_read_u32(&pdev->dev, "#thermal-sensor-cells",
+				     &priv->count)) {
+		/* register single cell if #thermal-sensor-cells is missing */
+		priv->count = 0;
+	}
+
 	priv->sensors = devm_kzalloc(&pdev->dev, priv->count *
 				     sizeof(struct ltq_thermal_sensor),
 				     GFP_KERNEL);
